### ISSUE TITLE
Logged in should always land me on my own shop 

### ISSUE THREAD

**Issue by cindyfirestone** [reactioncommerce/reaction#3184]
### Expected behavior
As a store operator with a shop on a marketplace, when I log into my shop it should land me on the shop associated with the account I logged in with. 

### Actual behavior
When I login, it lands me on the main marketplace, and I need to go into the dropdown in the upper left hand corner and select my shop.

### Steps to reproduce the behavior
1. Log into your shop
2. Land on main marketplace 
3. Go to upper left hand corner and select your store

![home](https://user-images.githubusercontent.com/22083616/32116019-d09f1740-bafd-11e7-8c8c-64d55111c4b2.png)

**Comment by speedingdeer** [comment]
same here

**Comment by kieckhafer** [comment]
@spencern 

https://github.com/reactioncommerce/reaction/blame/master/client/modules/core/main.js#L327

What are your thoughts on updating `hasShopSwitcherAccess` to only show the switcher if you are the overall Marketplace admin (Primary shop admin) instead of a user with dashboard access to any shop?

Something like this:

```
hasShopSwitcherAccess() {
    return hasAdminAccess(getPrimaryShopId());
  },
```

**Comment by spencern** [comment]
@kieckhafer we may need to show this shopSwitcher if you have admin access on more than one shop. E.g. writing Reaction wrapper around [Roles.getGroupsForUsers](http://alanning.github.io/meteor-roles/classes/Roles.html#method_getGroupsForUser)("admin") returns > 1 shopId.

Additionally, once we remove this, it becomes super important to verify that a user always has their activeShop set to the shop they have admin access for on login, and that it's unset on logout (in case of public computer, multi-user computer, etc)

**Comment by kieckhafer** [comment]
Merged in #3375
